### PR TITLE
Feature/product api

### PR DESCRIPTION
**Pull request by lvolzdev**
## 📄 Summary
- Get products owned
<br/>

## ✏️ Describe your changes
- Get Products owned by normal user (pagination)
- Get Products owned by corp user (pagination)
- add SearchConditions schema
<br/>

## 📍 Issue number and link
- #64 
- #66 
<br>

### CODE REVIEW

File: applications/product-application/src/main/java/com/pda/productapplication/dto/ProductDto.java [hunk 1]
```diff
@@ -13,22 +14,33 @@
...
         private String corpName;
+        @Schema(description = "Corporation logo image", example = "https://financial.pstatic.net/pie/common-bi/1.0.0/images/CD_SHINHAN_Profile.png")
         private String corpImage;
+        @Schema(description = "Card image (nullable)", example = "https://vertical.pstatic.net/vertical-cardad/creatives/SH/1408/SH_1408_20230213-143750_ver.png")
         private String cardImage;
+        @Schema(description = "Product tags", example = "['연회비지원', '관리비', '주유', '통신', '대형마트']")
         private List<String> tags;
+        @Schema(description = "Number of connected boards", example = "2")
         private int boardCount;
+        @Schema(description = "Created time", example = "2024-06-20 16:23:47.000000")
```
Comment (bkkmw) [inline applications/product-application/src/main/java/com/pda/productapplication/dto/ProductDto.java]: example에 T가 빠진 것 같습니다

File: applications/product-application/src/main/java/com/pda/productapplication/repository/ProductSpecification.java
Comment (bkkmw) [inline applications/product-application/src/main/java/com/pda/productapplication/repository/ProductSpecification.java:1]: LGTM

File: applications/product-application/src/main/java/com/pda/productapplication/service/ProductService.java [hunk 2]
```diff
@@ -10,12 +11,10 @@
...
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Service;
 
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
```
Comment (bkkmw) [inline applications/product-application/src/main/java/com/pda/productapplication/service/ProductService.java:17]: #68

File: applications/product-application/src/main/java/com/pda/productapplication/dto/ProductDto.java (hunk 1 above)
Comment (lvolzdev) [reply applications/product-application/src/main/java/com/pda/productapplication/dto/ProductDto.java]: 수정했습니다.

**Pull request merged**